### DISCUSSION TITLE
2.1.0 release

### DISCUSSION THREAD

**Discussion by eschutho** [apache/superset#23164] [General]
This is a discussion for feedback and testing of the 2.1.0 release candidates. The current RC is RC1. 

The release candidate:
https://dist.apache.org/repos/dist/dev/superset/2.1.0rc1/

Git tag for the release:
https://github.com/apache/superset/tree/2.1.0rc1

The Change Log for the release:
https://github.com/apache/superset/blob/2.1.0rc1/CHANGELOG.md

The Updating instructions for the release:
https://github.com/apache/superset/blob/2.1.0rc1/UPDATING.md

Please vote for this release via the mailing list voting thread https://lists.apache.org/thread/cm63dzp5t0gg3x01qbm5rxz5pz2x2h6s

If you have any questions or feedback, you can post them here or in the #apache-releases slack channel. For discussions about bugs found, please post here as we will be keeping track of all bugs that need to go into any subsequent RCs from this list or the voting threads.

This is the full list of cherries that will be going on top of the 2.1 master cut: https://github.com/apache/superset/pulls?q=is%3Apr++label%3Av2.1+

**Comment by paulvanharen** [comment]
I found some issues installing 2.1.0rc1 destroying the 'out of the box' experience. Please find my findings at #23181.

**Comment by sfirke** [comment]
Under 2.1.0rc1 I am still experiencing the issue where on a tabbed dashboard with any filters out of scope on the first tab, the charts just spin and don't load.  There are many issues for this bug, here's a discussion of it: https://github.com/apache/superset/discussions/21263#discussioncomment-3788816  I believe @codek confirmed this behavior in Slack.

**Reply by adimyth** [reply]
I am able to confirm that this issue persists. I tested out [apache/superset:2.1.0rc1 docker image](https://hub.docker.com/layers/apache/superset/2.1.0rc1/images/sha256-35ff5d8879641545cb23a4f7a16213a0bdf0b00111a7605e6c6862c3c33ae380?context=explore)

Is the solution to #21263 a part of this release? 

This is the one, I am looking forward to the most!

**Reply by eschutho** [reply]
Yes, all of the PRs that were created to fix the issue have been included in the 2.1 release. It looks like someone will need to submit a new PR for the fix.

**Reply by eschutho** [reply]
@lilykuang has volunteered to look into this issue

**Reply by sfirke** [reply]
This is solved in 2.1.0rc2 🙌

**Comment by sfirke** [comment]
There was a report on the dev listserve of a blank "Customize" tab on the Heatmap chart.  I can reproduce with 2.1.0rc1, see below.  But should there be any content on that Customize tab, or is the problem that the tab is included at all?  On 2.0.1 Heatmap chart I don't see a Customize tab at all.
![image](https://user-images.githubusercontent.com/7569808/221647380-cfb1eec3-1295-4fb8-8620-7cc3ce0ee0cb.png)

**Reply by eschutho** [reply]
@lilykuang has volunteered to look into this issue as well

**Reply by lilykuang** [reply]
fix for this https://github.com/apache/superset/pull/23243

**Comment by sfirke** [comment]
I've completed my testing.  I reported two minor bugs that seem to be regressions in 2.1.0rc1: #23197 and #23220.  They could probably be fixed in a 2.1.1 release IMO rather than holding up 2.1.0 if a fix is not currently available.

I added a note to anyone who can help debug the issue of some tabbed dashboards not loading when filters are out of scope: https://github.com/apache/superset/issues/20930#issuecomment-1446921659 and I'm happy to discuss that further, screenshare my examples, and/or QA a possible fix.

**Reply by eschutho** [reply]
Thank you for this testing, @sfirke!

**Reply by eschutho** [reply]
@kgabryje made a quick fix for #23220: https://github.com/apache/superset/pull/23237

**Reply by sfirke** [reply]
thanks @kgabryje - that is a _fast_ fix 😮   I'll plan to confirm the fix in 2.1.0rc2

**Comment by codek** [comment]
Fantastic! I'll also check rc2.

**Comment by mdeshmu** [comment]
can this be included in 2.1.0rc2 ?

https://github.com/apache/superset/pull/23139

**Reply by eschutho** [reply]
Yes, just added, thank you.

**Comment by eschutho** [comment]
This was mentioned on slack: https://github.com/apache/superset/issues/23001

**Comment by zhaorui2022** [comment]
Is it possible to include https://github.com/apache/superset/pull/20448? I tried it locally using docker-compose and it has fixed an issue we encountered mentioned in https://github.com/apache/superset/issues/23289 It is a large commit and there were merge conflicts. I ended accepting everything in this commit and it seems working fine for me when using it locally.

**Reply by zhaorui2022** [reply]
by the way, when I say there were merge conflicts, my base was version 2.0.1.

**Comment by Samira-El** [comment]
Hi,
I wanted to test out the new feature introduced by this [PR](https://github.com/apache/superset/pull/21931/files).

The [doc](https://github.com/apache/superset/blob/2.1.0rc1/docs/docs/installation/alerts-reports.mdx) says to add this: 

```python
from superset.reports.types import ReportScheduleExecutor
ALERT_REPORTS_EXECUTE_AS = [
    ReportScheduleExecutor.CREATOR_OWNER,
    ReportScheduleExecutor.CREATOR,
    ReportScheduleExecutor.MODIFIER_OWNER,
    ReportScheduleExecutor.MODIFIER,
    ReportScheduleExecutor.OWNER,
    ReportScheduleExecutor.SELENIUM,
]
```

But the enum `ReportScheduleExecutor` is no where to be found in the code.
It appears that this [PR](https://github.com/apache/superset/pull/22328) refactored the feature but the doc is outdated now.

**Reply by sfirke** [reply]
Looks like those docs just got updated in a PR today: https://github.com/apache/superset/pull/23323

**Comment by romankevich** [comment]
Is it possible to include this in 2.1.0 ?

- feat: Frontend tagging [#20876](https://github.com/apache/superset/pull/20876)
- feat: Add excel export [#22006](https://github.com/apache/superset/pull/22006)

**Comment by D3nn3** [comment]
Could we look into the following issue: https://github.com/apache/superset/issues/23218? I'd highly appreciate it :slightly_smiling_face:

**Comment by codek** [comment]
Yay whoop!

I'll try it out tomorrow.

**Comment by dttad** [comment]
Is drilldown feature supported in Apache Superset 2.1.0?

**Reply by mdeshmu** [reply]
Drill to Detail is available in 2.1.0. Drill By is yet to released.

**Comment by dat-linux** [comment]
Error: (sqlite3.OperationalError) no such table: ssh_tunnels
[SQL: SELECT ssh_tunnels.uuid AS ssh_tunnels_uuid, ssh_tunnels.created_on AS ssh_tunnels_created_on, ssh_tunnels.changed_on AS ssh_tunnels_changed_on, ssh_tunnels.extra_json AS ssh_tunnels_extra_json, ssh_tunnels.id AS ssh_tunnels_id, ssh_tunnels.database_id AS ssh_tunnels_database_id, ssh_tunnels.server_address AS ssh_tunnels_server_address, ssh_tunnels.server_port AS ssh_tunnels_server_port, ssh_tunnels.username AS ssh_tunnels_username, ssh_tunnels.password AS ssh_tunnels_password, ssh_tunnels.private_key AS ssh_tunnels_private_key, ssh_tunnels.private_key_password AS ssh_tunnels_private_key_password, ssh_tunnels.created_by_fk AS ssh_tunnels_created_by_fk, ssh_tunnels.changed_by_fk AS ssh_tunnels_changed_by_fk 
FROM ssh_tunnels 
WHERE ssh_tunnels.database_id = ?]
[parameters: (1,)]
(Background on this error at: https://sqlalche.me/e/14/e3q8)

**Reply by mdeshmu** [reply]
Seems like missing table for recently added ssh tunnels support for database connections. Can you perform “superset db upgrade” and see if that solves the problem.

**Reply by sandeeppennant** [reply]
I have the same issue, even after the superset db upgrade the issue remain's same.
Your suggestions/help is much appreciated

**Reply by sandeeppennant** [reply]
> Seems like missing table for recently added ssh tunnels support for database connections. Can you perform “superset db upgrade” and see if that solves the problem.

I have the same issue, even after the superset db upgrade the issue remain's same.
Your suggestions/help is much appreciated

**Comment by LeoDiep** [comment]
Hi, is the 2.1.0 update this freezing column function for pivot table. This would be massive useful for users when checking large tables or for who having small screen.
https://github.com/apache/superset/issues/8238

**Comment by Gurbrinder0031** [comment]
Hi. in 2.1.0 '{{ from_dttm }}'  is not working in dataset showing same in response and same for current user id.

**Comment by Gurbrinder0031** [comment]
![screenshot-superset mastersoftwaretechnologies com-2023 06 26-14_21_02](https://github.com/apache/superset/assets/19300278/adbd1b77-bfaf-412f-97e8-3b09676c36d5)
Saved and Discard buttons are not showing in apache superset2.1.0
